src/panels/UpdatePanel.php

```diff
@@ -59,23 +71,22 @@ public function getContactContent()
 
     public function getCurrentSilverStripeVersion()
     {
-        $updatePanelCache = SS_Cache::factory('DashboardUpdatePanel');
-        $result = $updatePanelCache->load('CurrentSilverStripeVersion');
-        if ($result) {
-            return UpdateVersion::from_version_string($result);
+        $updatePanelCache = Injector::inst()->get(CacheInterface::class . '.myCache');
+
+        if ($updatePanelCache->has('CurrentSilverStripeVersion')) {
+            return UpdateVersion::from_version_string($updatePanelCache->get('CurrentSilverStripeVersion'));
         }
 
-        $versions = explode(', ', Injector::inst()->get('LeftAndMain')->CMSVersion());
+        $versions = explode(', ', Injector::inst()->get('SilverStripe\Admin\LeftAndMain')->CMSVersion());
```

Comment:
We need to make sure we address the issues raised in #107 here too. I see you changed it to use "CMS: " but was that all we really needed to change to fix this?

Reply:
The reason I changed it to use CMS instead of Framework is because retrieving Framework in 4.0 came up with null for me. This change was purely to get things working. Do you think we should leave it as changed to check the CMS version? Or investigate why fetching Framework didn't work?

Reply:
I think from @aNickzz 's description, it sounds like the SilverStripe CMS module actually intentionally sets the Framework version to nothing. It might be exactly the same thing you saw in SS4.

I'm happy to just use the CMS version but I guess one thing we should probably do is a few more safety checks around the version number in case from some reason "CMS" is blank too.